sos/report/plugins/microovn.py

```diff
@@ -0,0 +1,152 @@
+        for table in tables:
+            cmds.append(f'{ovn_cmd} list {table}')
+
+    def setup(self):
+        # check if env is a clustered or non-clustered one
+        for pidfile in ['ovnnb_db.pid', 'ovnsb_db.pid',
+                        'ovn-northd.pid', 'ovn-controller.pid']:
+            self.add_copy_spec([
+                self.path_join('/var/snap/microovn/common/run/ovn', pidfile),
+            ])
+
+        ovn_controller_sock_path = self._find_sock(
+            self.ovn_sock_path, self.ovn_controller_sock_regex)
+
+        northd_sock_path = self._find_sock(self.ovn_sock_path,
+                                           self.ovn_northd_sock_regex)
+
+        # ovsdb nb/sb cluster status commands
+        cst = "cluster/status"
+        cos = "connection-status"
+        pfx = "microovn."
+        ovn_nb = "OVN_Northbound"
+        ovn_sb = "OVN_Southbound"
+        self.add_cmd_output([
+            f"{pfx}ovs-appctl -t {self.ovn_nbdb_sock_path} {cst} {ovn_nb}",
+            f"{pfx}ovs-appctl -t {self.ovn_sbdb_sock_path} {cst} {ovn_sb}",
+            f"{pfx}ovn-appctl -t {northd_sock_path} status",
+            f"{pfx}ovn-appctl -t {ovn_controller_sock_path} {cos}",
+        ], foreground=True, timeout=30
+        )
+
+        # Some user-friendly versions of DB output
+        nbctl_cmds = [
+            '{pfx}ovn-nbctl --no-leader-only show',
+            '{pfx}ovn-nbctl --no-leader-only get-ssl',
+            '{pfx}ovn-nbctl --no-leader-only get-connection',
+        ]
+
+        sbctl_cmds = [
+            '{pfx}ovn-sbctl --no-leader-only show',
+            '{pfx}ovn-sbctl --no-leader-only lflow-list',
+            '{pfx}ovn-sbctl --no-leader-only get-ssl',
+            '{pfx}ovn-sbctl --no-leader-only get-connection',
+        ]
+
+        # Database Output
+        nb_tables = self.get_tables_from_schema(self.path_join(
+            '/snap/microovn/current/share/ovn', 'ovn-nb.ovsschema'))
+        self.add_database_output(nb_tables, nbctl_cmds,
+                                 'microovn.ovn-nbctl --no-leader-only')
+        cmds = nbctl_cmds
+
+        sb_tables = self.get_tables_from_schema(self.path_join(
+            '/snap/microovn/current/share/ovn',
+            'ovn-sb.ovsschema'), ['Logical_Flow'])
+        self.add_database_output(sb_tables, sbctl_cmds,
+                                 'microovn.ovn-sbctl --no-leader-only')
+        cmds += sbctl_cmds
+
+        # If OVN is containerized, we need to run the above commands inside
+        # the container. Removing duplicates (in case there are) to avoid
+        # failing on collecting output to file on container running commands
+        cmds = list(set(cmds))
+        self.add_cmd_output(
+            cmds, foreground=True
+        )
+
+        for dbfile in ["ovnnb_db.db", "ovnsb_db.db"]:
+            path = '/var/snap/microovn/common/data/central/db'
+            dbfilepath = self.path_join(path, dbfile)
+            if os.path.exists(dbfilepath):
```

Comment:
Also check `self.path_exists()` here to account for containerized executions.

Reply:
Done. Will be coming with the next commit.